joblib/_parallel_backends.py

```diff
@@ -612,16 +614,8 @@ def __call__(self, *args, **kwargs):
             # interrupt processing for a KeyboardInterrupt
             raise WorkerInterrupt()
         except BaseException:
-            if PY27:
-                # Capture the traceback of the worker to make it part of
-                # the final exception message.
-                e_type, e_value, e_tb = sys.exc_info()
-                text = format_exc(e_type, e_value, e_tb, context=10,
-                                  tb_offset=1)
-                raise TransportableException(text, e_type)
```

Comment:
I think you can remove `TransportableException` everywhere. It looks like it was the only place where it was created no?

Reply:
I just did it in 584dd2b. But giving this a second thought, maybe we should deprecate it and not remove it.

Reply:
Indeed I did a quick search on github and there exists third-party code bases that import `joblib.my_exceptions.TransportableException`. So I think we should at least still keep the class there even if we do not use it anymore.

The problem is how to display a deprecation warning at import time. In Python 3.7 and later we can use module level `__getattr__` (https://www.python.org/dev/peps/pep-0562/) even if I am not 100% sure it would make it possible to raise the warning at import time.

Reply:
Yes you are right, it seems internal but it is not really.
So deprecation is probably necessary, but it seems tricky.

Reply:
It seems that it is not handled in this [dedicated library](https://github.com/tantale/deprecated) but if we find a good solution, this can probably be contributed there :)